uvicorn/protocols/http/httptools_impl.py

```diff
@@ -154,7 +155,10 @@ def data_received(self, data: bytes) -> None:
         try:
             self.parser.feed_data(data)
         except httptools.HttpParserError:
-            msg = "Invalid HTTP request received."
+            if self.header_too_big:
+                msg = "Header size exceeds limit"
```

Comment:
it sould be return http 431 Request Header Fields Too Large
https://www.rfc-editor.org/rfc/rfc6585#section-5

Reply:
Makes sense, thanks!

But I'm not sure if there's a straightforward way to raise or respond with status code other than 400. 

May be we need a helper function similar to send400response for other status codes.